core/modules/user/user.entity.inc

```diff
@@ -249,6 +249,14 @@ class UserStorageController extends EntityDatabaseStorageController {
       }
     }
 
+    if (valid_email_address($entity->name)) {
+      // Check to see if matching is required.
+      $user_email_match = config_get('system.core', 'user_email_match');
+      if ($user_email_match && ($entity->name != $entity->mail)) {
+        throw new EntityMalformedException('The username and email are both addresses that do not match.');
+      }
+    }
```

Comment:
We should keep this validation but just remove the `$user_email_match = config_get('system.core', 'user_email_match');` checking